tests/test_database_integration.py

```
@@ -0,0 +1,1024 @@
+        ]
+        
+        for scenario in error_scenarios:
+            # Create mock service with error injection
+            call_count = 0
+            
+            async def error_injected_search(query):
+                nonlocal call_count
+                call_count += 1
+                
+                if call_count <= 2:  # Fail first 2 attempts
+                    if scenario["error_type"] == "rate_limit":
```

Ruff (0.11.9)

Function definition does not bind loop variable `scenario`

(B023)